src/Microsoft.Identity.Client/Features/ConfidentialClient/ConfidentialClientApplication.cs

```diff
@@ -51,7 +51,7 @@ namespace Microsoft.Identity.Client
     /// A web app is the most common confidential client. The clientId is exposed through the web browser, but the secret is passed only in the back channel 
     /// and never directly exposed. For details see https://aka.ms/msal-net-client-applications
     /// </remarks>
-    public sealed class ConfidentialClientApplication : ClientApplicationBase, IConfidentialClientApplication, IConfidentialClientApplicationWithCertificate
+    public sealed class ConfidentialClientApplication : ClientApplicationBase, IConfidentialClientApplication, IConfidentialClientApplicationWithCertificate, IByRefreshToken
```

Comment:
Why do we need a new interface here?  can't this just be added to ConfidentialClientApplication?

Reply:
I would like to hide this added functionality behind an explicit interface for 2 reasons. 1: the number of methods exposed is already large and it's hard to keep the overview. 2: it's only needed in certain scenarios such as for migration or when integrating with other systems. I expect this to be functionality only used by few clients and we generally don't recommend people to keep the RT's them selves as that defeats one of the purposes of the library).

---
In reply to: [245778126](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/759#discussion_r245778126) [](ancestors = 245778126)